modules/apps/transfer/keeper/grpc_query_test.go

```diff
@@ -245,12 +245,34 @@ func (suite *KeeperTestSuite) TestEscrowAddress() {
 			},
 			true,
 		},
+		{
+			"failure - channel not found due to invalid portID",
+			func() {
+				req = &types.QueryEscrowAddressRequest{
+					PortId:    ibctesting.InvalidID,
+					ChannelId: ibctesting.FirstChannelID,
+				}
+			},
+			false,
+		},
+		{
+			"failure - channel not found due to invalid channelID",
+			func() {
+				req = &types.QueryEscrowAddressRequest{
+					PortId:    ibctesting.TransferPort,
+					ChannelId: ibctesting.InvalidID,
+				}
+			},
+			false,
+		},
 	}
```

Comment:
Should we add a test where both port ID and channel ID are valid, but no channel is found for those parameters? Because the tests will fail on `validategRPCRequest` because either port ID or channel ID is not valid, but we don't have a test case that fails if `HasChannel` returns false, right?

Reply:
I might have misunderstood it :) invalidID is equal to IDIsInvalid (https://github.com/cosmos/ibc-go/blob/2555a7c504a904064d659e4c1a3a74000887f73d/testing/values.go#L37) but when we check for its validity in validategRPCRequest (https://github.com/cosmos/ibc-go/blob/main/modules/core/24-host/validate.go#L39-L60) I don't think this value will cause that function to return false? The ID is not blank, doesn't contain slashes, it's within the range for length, and satisfies the regex: https://github.com/cosmos/ibc-go/blob/main/modules/core/24-host/validate.go#L28

I just assumed that "invalidID" meant something like "an ID that would neer match an existing port/channel ID" but maybe I was wrong :)

Reply:
To verify this I commented out the check for `HasChannel` and ran the test again, and got this failure:

```
--- FAIL: TestKeeperTestSuite (55.03s)
    --- FAIL: TestKeeperTestSuite/TestEscrowAddress (1.59s)
        --- FAIL: TestKeeperTestSuite/TestEscrowAddress/Case_failure_-_channel_not_found_due_to_invalid_portID (0.40s)
            grpc_query_test.go:287: 
                        Error Trace:    /workspaces/ibc-go/modules/apps/transfer/keeper/grpc_query_test.go:287
                                                                /go/pkg/mod/github.com/stretchr/testify@v1.9.0/suite/suite.go:115
                        Error:          An error is expected but got nil.
                        Test:           TestKeeperTestSuite/TestEscrowAddress/Case_failure_-_channel_not_found_due_to_invalid_portID
        --- FAIL: TestKeeperTestSuite/TestEscrowAddress/Case_failure_-_channel_not_found_due_to_invalid_channelID (0.43s)
            grpc_query_test.go:287: 
                        Error Trace:    /workspaces/ibc-go/modules/apps/transfer/keeper/grpc_query_test.go:287
                                                                /go/pkg/mod/github.com/stretchr/testify@v1.9.0/suite/suite.go:115
                        Error:          An error is expected but got nil.
                        Test:           TestKeeperTestSuite/TestEscrowAddress/Case_failure_-_channel_not_found_due_to_invalid_channelID
```

which leads me to think that we are testing for the correct path here?

Reply:
(that being said, this means that we can -and should- probably add a test for `validateGRPCRequest`, though maybe this can be delayed until we maybe consolidate it into a single location, as discussed here https://github.com/cosmos/ibc-go/pull/6059#discussion_r1539199138)

Reply:
Oh, yes, my bad. I misinterpreted the function :)

Yes, sounds good to wait to see where that function lands and test it once there.

Reply:
Although to come back to [Jim's point](https://github.com/cosmos/ibc-go/pull/6059#discussion_r1539215613), maybe it is worth to add a simple test case for both port ID empty and channel ID empty.

Reply:
Sounds good! I removed one of the two "channel not found" test because they were actually both just testing the same line (`hasChannel() == false`) and added one for an empty portID and one for an empty channelID

Reply:
Great, thank you!